lib/decoders/layer.js

```diff
@@ -73,7 +77,16 @@ class Layer extends Duplex {
   }
 
   get name() {
-    return `layer_${this._position}`;
+    return this._spec.name || `layer_${this._position}`;
```

Comment:
What's with all the underscores.  ...What do they mean in a trailing position? 

Do we name private variables with underscores in the frontend?  Because it's not something we do in other backend code.